homeassistant/components/vizio/media_player.py

```diff
@@ -217,20 +218,20 @@ async def async_update(self) -> None:
                         VIZIO_AUDIO_SETTINGS, VIZIO_SOUND_MODE
                     )
             else:
-                self._supported_commands ^= SUPPORT_SELECT_SOUND_MODE
+                # Explicitly remove SUPPORT_SELECT_SOUND_MODE from supported features
+                self._supported_commands &= ~SUPPORT_SELECT_SOUND_MODE
 
         input_ = await self._device.get_current_input(log_api_exception=False)
-        if input_ is not None:
+        if input_:
             self._current_input = input_
 
-        if not self._available_inputs:
-            inputs = await self._device.get_inputs_list(log_api_exception=False)
+        inputs = await self._device.get_inputs_list(log_api_exception=False)
 
-            # If no inputs returned, end update
-            if not inputs:
-                return
+        # If no inputs returned, end update
+        if not inputs:
+            return
 
-            self._available_inputs = [input_.name for input_ in inputs]
+        self._available_inputs = [input_.name for input_ in inputs]
```

Comment:
I think you sort of got my comment backward. my intention was:

```
inputs = await self._device.get_inputs_list(log_api_exception=False)
if inputs is None:
  inputs = []

self._available_inputs = [input_.name for input_ in inputs]
```

Now you are leaving self._available_inputs set to whatever it happened to be before. Ie you leave stale data around.

Reply:
Ah I see. OK will update

Reply:
I'm okey with that (even thou i would normally see empty list a normal non error case, but that would depend on the lib). But it's nicer to be more explicit than rather than return in that case. You are skipping all code below the return. It's easy to forget a return smack in the middle of a function.

Reply:
The return is purposefully there. The only reason an API call will fail is if the API is unavailable, either due to network connectivity loss or because the device is no longer responding., so there is no value in continuing to make API calls in this update pass if that call fails. BTW this used to be handled a different way but it was suggested to do the return instead to make the code more readable. I don't think there's an issue here but let me know if I'm missing something